src/umpire/strategy/FixedPool.cpp

```diff
@@ -105,7 +105,6 @@ FixedPool::allocInPool(Pool& p) noexcept
     }
   }
 
-  UMPIRE_ASSERT("FixedPool: Logic error in allocate");
```

Comment:
Why does this assertion no longer need to exist?

Reply:
Good catch. The UMPIRE_ASSERT placed there was malformed/dumb, it always failed because there isn't a condition. I tried switching it to an UMPIRE_ERROR, but the compiler spits out an error telling me the code will always fail at that point. It's not taking into account that it should really never execute that line. I was going to try and fix the compiler error later before merging.

Reply:
It sounds like you are asserting that this line should never be executed, which, in turn, sounds like a greate place for an UMPIRE_ASSERT

Reply:
`assert()` has a condition, which this doesn't. It should be an error-type macro.